app/controllers/submissions_controller.rb

```diff
@@ -436,6 +436,12 @@ def view
     @problemReleased = @submission.scores.pluck(:released).all?
 
     @annotations = @submission.annotations.to_a
+    unless @submission.group_key.empty?
+      group_submissions = @submission.group_associated_submissions
+      group_submissions.each do |group_submission|
+        @annotations += group_submission.annotations.to_a
```

Comment:
You are appending to `@annotations` which already has  `@submission.annotations.to_a` assigned to it.
Would that lead to the original submission having double the annotations? I don't see that when testing however.

Reply:
`group_associated_submissions` specifically gets all submissions _except_ the current submission (`Submission.where(group_key: group_key).where.not(id: id)`), so we do not duplicate annotations for the original submission.